### ISSUE TITLE
How to retrieve data after getting response

### ISSUE THREAD

**Issue by Blackening999** [AuthorizeNet/sample-code-ruby#25]
https://github.com/AuthorizeNet/sample-code-ruby/blob/master/CustomerProfiles/get-customer-payment-profile.rb

It isn't clear how to "unzip" response and then retrieve data on back-end for further transaction like ARB, doesn't matter actually.
I've tried to do:

```ruby
if response.messages.resultCode == MessageTypeEnum::Ok
     #HERE
      GetCustomerPaymentProfileResponse.new(nil, response.messages, nil, response.paymentProfile)
    else
      puts response.messages.messages[0].text
      raise "Failed to get payment profile information with id #{request.customerPaymentProfileId}"
    end
```

but I can't get any info from it :(

**Comment by namanbansal** [comment]
Hi @Blackening999 ,

I have updated the sample code to print some to the data from response. Have a look [here](https://github.com/namanbansal/sample-code-ruby/commit/5f52bf50853a5c01baa5451e1cb0dd2e28818ebd).
You will also have to incorporate changes from [PR](https://github.com/AuthorizeNet/sdk-ruby/pull/71) into your sdk and build the gem again.

Hope this helps!

**Comment by Blackening999** [comment]
Hi @namanbansal !

Thanks for support and help!

**Comment by CraftyBeer** [comment]
I've been running into the same problem. I can't tell if this just isn't built out fully or if the documentation is lacking.

response.paymentProfile is returning an object with a string of the payment profile data. I'm looking for objects that I can use to create things like recurring payments using a users CreditCard object. There's no way to even parse the payment profile data since the string portion of the object has no delimiters.

I've been using puts YAML::dump(response) to take a look into the responses.

**Comment by Blackening999** [comment]
btw, when PR is supposed to be merged into master?

**Comment by Blackening999** [comment]
@namanbansal 

and why do you use "CustomerPaymentProfileMaskedType" for casting payment object and not usual "CustomerProfilePaymentType" ?

In future it isn't possible to retrieve cc data from payment profile type :(

**Comment by Blackening999** [comment]
@brianmc 

The added sample code don't showing the full CC number and I can't reuse profile info for subscription (in my case). I have to force user enter data again and again which isn't good UX at all

**Comment by brianmc** [comment]
Changes should be merged into SDK master today and hopefully released later today.  

RE: using the profile to create a subscription, we will be adding that functionality to our createSubscription in the very, very near future.

RE: CustomerPaymentProfileMaskedType the reason we use this type is because the new SDK model uses the API schema to generate the language objects (by autogenerating the SDK we're able to keep it up-to-date more easily and eradicate object mapping coding errors).

**Comment by JimmyRPaul** [comment]
@brianmc @namanbansal - Thanks a lot guys.

I'm looking to do the exact same thing as @Blackening999. I need to reuse profile credit card info for a subscription. Thanks again and keep us updated.

**Comment by Blackening999** [comment]
While we all here it's also interesting one case I haven't found in docs.
When subscription will end after cancellation?
At periods end or at the moment of action?

I know it's off topic, but please, help me with that

Thanks a lot guys for what you're doing! I haven't found any support except here!

**Comment by brianmc** [comment]
If a subscription is cancelled it will end immediately with not further charges.

I'm going to close this issue but watch out for an update on the developer blog about this update to our CreateSubscription API.  Also follow @AuthorizeNetDev to get news of API updates.

Brian

**Comment by brianmc** [comment]
So the changes for PaymentProfile Response (original issues - not the CreateSubscription we were also discussing) are merged into master, tagged release 1.8.7 (https://github.com/AuthorizeNet/sdk-ruby/releases/tag/1.8.7) and published new gem https://rubygems.org/gems/authorizenet/versions/1.8.7

Thanks everyone,

Brian
